File: velolabs_repos/lattis_android/app/src/main/java/com/lattis/ellipse/presentation/UIThread.java
package com.lattis.ellipse.presentation;

import com.lattis.ellipse.domain.executor.PostExecutionThread;

import javax.inject.Inject;
import javax.inject.Singleton;

import io.reactivex.Scheduler;
import io.reactivex.android.schedulers.AndroidSchedulers;


@Singleton
public class UIThread implements PostExecutionThread {

    @Inject
    public UIThread() {
    }

    @Override public Scheduler getScheduler() {
        return AndroidSchedulers.mainThread();
    }
}

File: velolabs_repos/operator-android/app/src/main/java/io/lattis/operator/executor/UiThread.kt
package io.lattis.operator.executor

import io.lattis.domain.executor.PostExecutionThread
import io.reactivex.Scheduler
import io.reactivex.android.schedulers.AndroidSchedulers
import javax.inject.Inject

class UiThread @Inject constructor() : PostExecutionThread {
    override val scheduler: Scheduler = AndroidSchedulers.mainThread()
}